task/sql-tasks.js

```diff
@@ -280,7 +498,34 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT DISTINCT
+        t_Cust.CompanyName,
+        t_Prod.ProductName,
+        t_OrderD.UnitPrice AS PricePerItem
+    FROM
+        Customers AS t_Cust
+            JOIN
+        Orders AS t_Order ON t_Order.CustomerID = t_Cust.CustomerID
+            JOIN
+        OrderDetails AS t_OrderD ON t_OrderD.OrderID = t_Order.OrderID
+            JOIN
+        Products AS t_Prod ON t_OrderD.ProductID = t_Prod.ProductID
+    WHERE
+        t_OrderD.UnitPrice = (SELECT 
+                MAX(t1_OrderD.UnitPrice)
+            FROM
+                Customers AS t1_Cust
+                    JOIN
+                Orders AS t1_Order ON t1_Order.CustomerID = t1_Cust.CustomerID
+                    JOIN
+                OrderDetails AS t1_OrderD ON t1_Order.OrderID = t1_OrderD.OrderID
+            WHERE
+                t1_Cust.CompanyName = t_Cust.CompanyName)

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.